pkg/identity/cache/allocator.go

```diff
@@ -110,12 +113,15 @@ type IdentityAllocatorOwner interface {
 	GetNodeSuffix() string
 }
 
-// InitIdentityAllocator creates the the identity allocator. Only the
-// first invocation of this function will have an effect.  Caller must
-// have initialized well known identities before calling this (by
-// calling identity.InitWellKnownIdentities()). Returns a channel which is
-// closed when initialization of the allocator is completed.
-func InitIdentityAllocator(owner IdentityAllocatorOwner) <-chan struct{} {
+// InitIdentityAllocator creates the the identity allocator. Only the first
+// invocation of this function will have an effect. The Caller must have
+// initialized well known identities before calling this (by calling
+// identity.InitWellKnownIdentities()).
+// client and identityStore are only used by the CRD identity allocator,
+// currently, and identityStore may be nil.
+// Returns a channel which is closed when initialization of the allocator is
+// completed.
+func InitIdentityAllocator(owner IdentityAllocatorOwner, client clientset.Interface, identityStore cache.Store) <-chan struct{} {
```

Comment:
I'm not a fan of pulling the K8s dependency into this package... ideally we'd be able to hide it behind an interface again, but that's a lot of work. Can you put a TODO here for this?